storelogs/main.go

```diff
@@ -0,0 +1,39 @@
+package storelogs
+
+import (
+	"container/ring"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+type LogsStore struct {
+	log   *ring.Ring
+	entry *logrus.Entry
+}
+
+func (l *LogsStore) SetUp(countLogs int, entry *logrus.Entry) {
+	l.log = ring.New(countLogs)
+	l.entry = entry
+}
+
+func (l *LogsStore) SaveLog(log string) {
+	dt := time.Now()
+	log = dt.Format("01-02-2006 15:04:05") + " " + l.entry.Level.String() + ": " + log
+	for _, v := range l.entry.Data {
+		log = log + fmt.Sprintf("    %v", v)
```

Comment:
1 solution make slice of strings and then join
2 solution by any buffer
in `log = log + fmt.Sprintf(" %v", v)` too many garbage https://gosamples.dev/concatenate-strings/

Reply:
changed to buffer